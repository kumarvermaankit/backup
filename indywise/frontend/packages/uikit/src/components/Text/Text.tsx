import * as React from 'react';
import styled from 'styled-components';

export type TextType =
  | 'h1'
  | 'h2'
  | 'h3'
  | 'h4'
  | 'hl'
  | 'title'
  | 'subtitle'
  | 'body'
  | 'small'
  | 'paragraph';

export interface ITextProps {
  /**
   * Type of text.
   *
   * @example `h1` for heading
   *
   * @example `hl` for large heading
   *
   * @default p
   */
  type?: TextType;

  /**
   * Text should be bold or not.
   *
   * @example `true` for bold text.
   *
   * @example `false` for normal text.
   */
  bold?: boolean;

  /**
   * Text should be underlined or not.
   *
   * @example `true` for underlined text.
   *
   * @example `false` for normal text.
   */
  underline?: boolean;

  /**
   * Color to override the default color.
   *
   * @example `#f3f3f3` or `green`
   */
  color?: string;

  /**
   * Size of text.
   *
   * @example `24px`
   */
  size?: string;

  /**
   * Custom CSS styling.
   */
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  style?: any;
}

export const Text: React.FC<ITextProps> = (props) => {
  const bold = props.bold || false;
  const style = props.style || {};
  if (props.underline) {
    style['text-decoration'] = 'underline';
  }
  if (props.color) {
    style['color'] = props.color;
  }

  switch (props.type) {
    case 'hl':
      return (
        <HL style={style} bold={bold} size={props.size}>
          {props.children}
        </HL>
      );

    case 'h1':
      return (
        <H1 style={style} bold={bold} size={props.size}>
          {props.children}
        </H1>
      );

    case 'h2':
      return (
        <H2 style={style} bold={bold} size={props.size}>
          {props.children}
        </H2>
      );

    case 'h3':
      return (
        <H3 style={style} bold={bold} size={props.size}>
          {props.children}
        </H3>
      );

    case 'h4':
      return (
        <H4 style={style} bold={bold} size={props.size}>
          {props.children}
        </H4>
      );

    case 'title':
      return (
        <Title style={style} bold={bold} size={props.size}>
          {props.children}
        </Title>
      );

    case 'subtitle':
      return (
        <SubTitle style={style} bold={bold} size={props.size}>
          {props.children}
        </SubTitle>
      );

    case 'body':
      return (
        <Body style={style} bold={bold} size={props.size}>
          {props.children}
        </Body>
      );

    case 'small':
      return (
        <Small style={style} bold={bold} size={props.size}>
          {props.children}
        </Small>
      );

    case 'paragraph':
      return (
        <StyledText style={style} bold={bold} size={props.size}>
          {props.children}
        </StyledText>
      );

    default:
      return (
        <StyledText style={style} bold={bold} size={props.size}>
          {props.children}
        </StyledText>
      );
  }
};

const HL = styled.h1<{
  bold: boolean;
  size: string | undefined;
}>`
  margin: 0;
  font-style: normal;
  font-size: ${(props) => props.size || '80px'};
  font-family: 'Mulish', sans-serif;
  font-weight: 900;
  line-height: 100px;
  color: #0c3559;
`;

const H1 = styled.h1<{
  bold: boolean;
  size: string | undefined;
}>`
  margin: 0;
  font-size: ${(props) => props.size || '64px'};
  font-style: normal;
  font-family: 'Mulish', sans-serif;
  font-weight: 900;
  line-height: 80px;
  color: #0c3559;
`;

const H2 = styled.h2<{
  bold: boolean;
  size: string | undefined;
}>`
  margin: 0;
  font-style: normal;
  font-size: ${(props) => props.size || '48px'};
  font-family: 'Mulish', sans-serif;
  font-weight: ${(props) => (props.bold ? '900' : '700')};
  line-height: 60px;
  color: #0c3559;
`;

const H3 = styled.h3<{
  bold: boolean;
  size: string | undefined;
}>`
  margin: 0;
  font-style: normal;
  font-size: ${(props) => props.size || '32px'};
  font-family: 'Mulish', sans-serif;
  font-weight: ${(props) => (props.bold ? '900' : '700')};
  line-height: 40px;
  color: #0c3559;
`;

const H4 = styled.h4<{
  bold: boolean;
  size: string | undefined;
}>`
  margin: 0;
  font-style: normal;
  font-size: ${(props) => props.size || '24px'};
  font-family: 'Mulish', sans-serif;
  font-weight: ${(props) => (props.bold ? '900' : '700')};
  line-height: 30px;
  color: #0c3559;
`;

const Title = styled.h1<{
  bold: boolean;
  size: string | undefined;
}>`
  margin: 0;
  font-style: normal;
  font-size: ${(props) => props.size || '1.25rem'};
  font-family: 'Roboto', sans-serif;
  font-weight: ${(props) => (props.bold ? 'bold' : 'normal')};
  line-height: 30px;
  color: #0c3559;
`;

const SubTitle = styled.h2<{
  bold: boolean;
  size: string | undefined;
}>`
  margin: 0;
  font-style: normal;
  font-size: ${(props) => props.size || '16px'};
  font-family: 'Roboto', sans-serif;
  font-weight: ${(props) => (props.bold ? 'bold' : 'normal')};
  line-height: 24px;
  color: #0c3559;
`;

const Body = styled.p<{
  bold: boolean;
  size: string | undefined;
}>`
  margin: 0;
  font-style: normal;
  font-size: ${(props) => props.size || '16px'};
  font-family: 'Roboto', sans-serif;
  font-weight: ${(props) => (props.bold ? 'bold' : 'normal')};
  line-height: 19px;
  color: #0c3559;
`;

const Small = styled.p<{
  bold: boolean;
  size: string | undefined;
}>`
  margin: 0;
  font-style: normal;
  font-size: ${(props) => props.size || '12px'};
  font-family: 'Roboto', sans-serif;
  font-weight: ${(props) => (props.bold ? 'bold' : 'normal')};
  line-height: 18px;
  color: #0c3559;
`;

const StyledText = styled.p<{
  bold: boolean;
  size: string | undefined;
}>`
  margin: 0;
  font-style: normal;
  font-family: 'Roboto', sans-serif;
  font-weight: ${(props) => (props.bold ? 'bold' : 'normal')};
  font-size: ${(props) => props.size || '16px'};
  line-height: 18px;
  color: #0c3559;
`;
