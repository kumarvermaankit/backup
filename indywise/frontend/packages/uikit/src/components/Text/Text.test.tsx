import * as React from 'react';
import { create } from 'react-test-renderer';

import { Text } from './Text';

describe('Text', () => {
  it('H1', () => {
    const component = create(<Text type="h1">Sample Text</Text>).toJSON();

    expect(component).toMatchSnapshot();
  });

  it('H1 bold', () => {
    const component = create(
      <Text type="h1" bold={true}>
        Sample Text
      </Text>
    ).toJSON();

    expect(component).toMatchSnapshot();
  });

  it('H1 underline', () => {
    const component = create(
      <Text type="h1" underline={true}>
        Sample Text
      </Text>
    ).toJSON();

    expect(component).toMatchSnapshot();
  });

  it('H2', () => {
    const component = create(<Text type="h2">Sample Text</Text>).toJSON();

    expect(component).toMatchSnapshot();
  });

  it('H2 bold', () => {
    const component = create(
      <Text type="h2" bold={true}>
        Sample Text
      </Text>
    ).toJSON();

    expect(component).toMatchSnapshot();
  });

  it('H2 underline', () => {
    const component = create(
      <Text type="h2" underline={true}>
        Sample Text
      </Text>
    ).toJSON();

    expect(component).toMatchSnapshot();
  });

  it('H3', () => {
    const component = create(<Text type="h3">Sample Text</Text>).toJSON();

    expect(component).toMatchSnapshot();
  });

  it('H3 bold', () => {
    const component = create(
      <Text type="h3" bold={true}>
        Sample Text
      </Text>
    ).toJSON();

    expect(component).toMatchSnapshot();
  });

  it('H3 underline', () => {
    const component = create(
      <Text type="h3" underline={true}>
        Sample Text
      </Text>
    ).toJSON();

    expect(component).toMatchSnapshot();
  });

  it('H4', () => {
    const component = create(<Text type="h4">Sample Text</Text>).toJSON();

    expect(component).toMatchSnapshot();
  });

  it('H4 bold', () => {
    const component = create(
      <Text type="h4" bold={true}>
        Sample Text
      </Text>
    ).toJSON();

    expect(component).toMatchSnapshot();
  });

  it('H4 underline', () => {
    const component = create(
      <Text type="h4" underline={true}>
        Sample Text
      </Text>
    ).toJSON();

    expect(component).toMatchSnapshot();
  });

  it('HL', () => {
    const component = create(<Text type="hl">Sample Text</Text>).toJSON();

    expect(component).toMatchSnapshot();
  });

  it('HL bold', () => {
    const component = create(
      <Text type="hl" bold={true}>
        Sample Text
      </Text>
    ).toJSON();

    expect(component).toMatchSnapshot();
  });

  it('HL underline', () => {
    const component = create(
      <Text type="hl" underline={true}>
        Sample Text
      </Text>
    ).toJSON();

    expect(component).toMatchSnapshot();
  });

  it('Body', () => {
    const component = create(<Text type="body">Sample Text</Text>).toJSON();

    expect(component).toMatchSnapshot();
  });

  it('Body bold', () => {
    const component = create(
      <Text type="body" bold={true}>
        Sample Text
      </Text>
    ).toJSON();

    expect(component).toMatchSnapshot();
  });

  it('Body underline', () => {
    const component = create(
      <Text type="body" underline={true}>
        Sample Text
      </Text>
    ).toJSON();

    expect(component).toMatchSnapshot();
  });

  it('Paragraph', () => {
    const component = create(
      <Text type="paragraph">Sample Text</Text>
    ).toJSON();

    expect(component).toMatchSnapshot();
  });

  it('Paragraph bold', () => {
    const component = create(
      <Text type="paragraph" bold={true}>
        Sample Text
      </Text>
    ).toJSON();

    expect(component).toMatchSnapshot();
  });

  it('Paragraph underline', () => {
    const component = create(
      <Text type="paragraph" underline={true}>
        Sample Text
      </Text>
    ).toJSON();

    expect(component).toMatchSnapshot();
  });

  it('Title', () => {
    const component = create(<Text type="title">Sample Text</Text>).toJSON();

    expect(component).toMatchSnapshot();
  });

  it('Title bold', () => {
    const component = create(
      <Text type="title" bold={true}>
        Sample Text
      </Text>
    ).toJSON();

    expect(component).toMatchSnapshot();
  });

  it('Title underline', () => {
    const component = create(
      <Text type="title" underline={true}>
        Sample Text
      </Text>
    ).toJSON();

    expect(component).toMatchSnapshot();
  });

  it('Small', () => {
    const component = create(<Text type="small">Sample Text</Text>).toJSON();

    expect(component).toMatchSnapshot();
  });

  it('Small bold', () => {
    const component = create(
      <Text type="small" bold={true}>
        Sample Text
      </Text>
    ).toJSON();

    expect(component).toMatchSnapshot();
  });

  it('Small underline', () => {
    const component = create(
      <Text type="small" underline={true}>
        Sample Text
      </Text>
    ).toJSON();

    expect(component).toMatchSnapshot();
  });

  it('Subtitle', () => {
    const component = create(<Text type="subtitle">Sample Text</Text>).toJSON();

    expect(component).toMatchSnapshot();
  });

  it('Subtitile bold', () => {
    const component = create(
      <Text type="subtitle" bold={true}>
        Sample Text
      </Text>
    ).toJSON();

    expect(component).toMatchSnapshot();
  });

  it('Subtitle underline', () => {
    const component = create(
      <Text type="subtitle" underline={true}>
        Sample Text
      </Text>
    ).toJSON();

    expect(component).toMatchSnapshot();
  });

  it('Size Color', () => {
    const component = create(
      <Text size="24px" color="red">
        Sample Text
      </Text>
    ).toJSON();

    expect(component).toMatchSnapshot();
  });

  it('Styled', () => {
    const component = create(
      <Text
        style={{
          color: '#333',
          fontSize: '40px'
        }}
      >
        Sample Text
      </Text>
    ).toJSON();

    expect(component).toMatchSnapshot();
  });
});
