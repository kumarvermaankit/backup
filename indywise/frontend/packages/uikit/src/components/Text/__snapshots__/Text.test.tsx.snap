// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text Body 1`] = `
<p
  className=""
  style={Object {}}
>
  Sample Text
</p>
`;

exports[`Text Body bold 1`] = `
<p
  className=""
  style={Object {}}
>
  Sample Text
</p>
`;

exports[`Text Body underline 1`] = `
<p
  className=""
  style={
    Object {
      "text-decoration": "underline",
    }
  }
>
  Sample Text
</p>
`;

exports[`Text H1 1`] = `
<h1
  className=""
  style={Object {}}
>
  Sample Text
</h1>
`;

exports[`Text H1 bold 1`] = `
<h1
  className=""
  style={Object {}}
>
  Sample Text
</h1>
`;

exports[`Text H1 underline 1`] = `
<h1
  className=""
  style={
    Object {
      "text-decoration": "underline",
    }
  }
>
  Sample Text
</h1>
`;

exports[`Text H2 1`] = `
<h2
  className=""
  style={Object {}}
>
  Sample Text
</h2>
`;

exports[`Text H2 bold 1`] = `
<h2
  className=""
  style={Object {}}
>
  Sample Text
</h2>
`;

exports[`Text H2 underline 1`] = `
<h2
  className=""
  style={
    Object {
      "text-decoration": "underline",
    }
  }
>
  Sample Text
</h2>
`;

exports[`Text H3 1`] = `
<h3
  className=""
  style={Object {}}
>
  Sample Text
</h3>
`;

exports[`Text H3 bold 1`] = `
<h3
  className=""
  style={Object {}}
>
  Sample Text
</h3>
`;

exports[`Text H3 underline 1`] = `
<h3
  className=""
  style={
    Object {
      "text-decoration": "underline",
    }
  }
>
  Sample Text
</h3>
`;

exports[`Text H4 1`] = `
<h4
  className=""
  style={Object {}}
>
  Sample Text
</h4>
`;

exports[`Text H4 bold 1`] = `
<h4
  className=""
  style={Object {}}
>
  Sample Text
</h4>
`;

exports[`Text H4 underline 1`] = `
<h4
  className=""
  style={
    Object {
      "text-decoration": "underline",
    }
  }
>
  Sample Text
</h4>
`;

exports[`Text HL 1`] = `
<h1
  className=""
  style={Object {}}
>
  Sample Text
</h1>
`;

exports[`Text HL bold 1`] = `
<h1
  className=""
  style={Object {}}
>
  Sample Text
</h1>
`;

exports[`Text HL underline 1`] = `
<h1
  className=""
  style={
    Object {
      "text-decoration": "underline",
    }
  }
>
  Sample Text
</h1>
`;

exports[`Text Paragraph 1`] = `
<p
  className=""
  style={Object {}}
>
  Sample Text
</p>
`;

exports[`Text Paragraph bold 1`] = `
<p
  className=""
  style={Object {}}
>
  Sample Text
</p>
`;

exports[`Text Paragraph underline 1`] = `
<p
  className=""
  style={
    Object {
      "text-decoration": "underline",
    }
  }
>
  Sample Text
</p>
`;

exports[`Text Size Color 1`] = `
<p
  className=""
  size="24px"
  style={
    Object {
      "color": "red",
    }
  }
>
  Sample Text
</p>
`;

exports[`Text Small 1`] = `
<p
  className=""
  style={Object {}}
>
  Sample Text
</p>
`;

exports[`Text Small bold 1`] = `
<p
  className=""
  style={Object {}}
>
  Sample Text
</p>
`;

exports[`Text Small underline 1`] = `
<p
  className=""
  style={
    Object {
      "text-decoration": "underline",
    }
  }
>
  Sample Text
</p>
`;

exports[`Text Styled 1`] = `
<p
  className=""
  style={
    Object {
      "color": "#333",
      "fontSize": "40px",
    }
  }
>
  Sample Text
</p>
`;

exports[`Text Subtitile bold 1`] = `
<h2
  className=""
  style={Object {}}
>
  Sample Text
</h2>
`;

exports[`Text Subtitle 1`] = `
<h2
  className=""
  style={Object {}}
>
  Sample Text
</h2>
`;

exports[`Text Subtitle underline 1`] = `
<h2
  className=""
  style={
    Object {
      "text-decoration": "underline",
    }
  }
>
  Sample Text
</h2>
`;

exports[`Text Title 1`] = `
<h1
  className=""
  style={Object {}}
>
  Sample Text
</h1>
`;

exports[`Text Title bold 1`] = `
<h1
  className=""
  style={Object {}}
>
  Sample Text
</h1>
`;

exports[`Text Title underline 1`] = `
<h1
  className=""
  style={
    Object {
      "text-decoration": "underline",
    }
  }
>
  Sample Text
</h1>
`;
